### PR TITLE
English translation

### PR DESCRIPTION
**Pull request by emkookmer**
These updates by @benfiglin will translate the code to English as requested in #3

### CODE REVIEW

**Comment by benfiglin** [comment]
Hmm I actually wasn't planning on creating a PR for that yet. I had to do a quick and dirty patch and should have done it in a separate branch since there are other things in there that do not belong to the English translation.

File: HomeAssistant.js [hunk 1]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
                 delete mergeConfig[key]
             }
         })
-        this.publish(discovery_topic, mergeConfig)
-        this.node.status({ fill: "green", shape: "ring", text: `更新配置：${name}` });
+        this.publish(discovery_topic, mergeConfig, "", { retain: true })
+        this.node.status({ fill: "green", shape: "ring", text: `Update configuration：${name}` });
     }
 
     // 状态
```
Comment (emkookmer) [inline HomeAssistant.js:119]: missing translation : "state"

File: HomeAssistant.js [hunk 2]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_state(data) {
-        this.publish(this.config.state_topic, data, "更新状态")
+        this.publish(this.config.state_topic, data, "Update state")
     }
 
     // 属性
```
Comment (emkookmer) [inline HomeAssistant.js:124]: missing translation : "Attributes"

File: HomeAssistant.js [hunk 3]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_attributes(data) {
-        this.publish(this.config.json_attr_t, data, "更新属性")
+        this.publish(this.config.json_attr_t, data, "Update attribute")
     }
 
     // 当前温度
```
Comment (emkookmer) [inline HomeAssistant.js:129]: missing translation : "Current Temperature"

File: HomeAssistant.js [hunk 4]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_current_temperature(data) {
-        this.publish(this.config.current_temperature_topic, data, "更新当前温度")
+        this.publish(this.config.current_temperature_topic, data, "Update the current temperature")
     }
 
     // 当前温度
```
Comment (emkookmer) [inline HomeAssistant.js:134]: missing translation  or remove comment

File: HomeAssistant.js [hunk 5]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_temperature(data) {
-        this.publish(this.config.temperature_state_topic, data, "更新温度")
+        this.publish(this.config.temperature_state_topic, data, "Update temperature")
     }
 
     // 目标湿度
```
Comment (emkookmer) [inline HomeAssistant.js:139]: missing translation : "Target Humidity"

File: HomeAssistant.js [hunk 6]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_target_humidity(data) {
-        this.publish(this.config.target_humidity_state_topic, data, "更新目标湿度")
+        this.publish(this.config.target_humidity_state_topic, data, "Update the target humidity")
     }
 
     // 效果
```
Comment (emkookmer) [inline HomeAssistant.js:144]: missing translation : "Effect"

File: HomeAssistant.js [hunk 7]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_effect(data) {
-        this.publish(this.config.effect_state_topic, data, "更新特效")
+        this.publish(this.config.effect_state_topic, data, "Update effects")
     }
 
     // 摆动
```
Comment (emkookmer) [inline HomeAssistant.js:149]: missing translation : "Swing / Oscillation"

File: HomeAssistant.js [hunk 8]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_oscillation(data) {
-        this.publish(this.config.oscillation_state_topic, data, "更新摆动")
+        this.publish(this.config.oscillation_state_topic, data, "Update oscillation")
     }
 
     // 百分比
```
Comment (emkookmer) [inline HomeAssistant.js:154]: missing translation : "Percentage"

File: HomeAssistant.js [hunk 9]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_percentage(data) {
-        this.publish(this.config.percentage_state_topic, data, "更新百分比")
+        this.publish(this.config.percentage_state_topic, data, "Update percentage")
     }
 
     // 预设模式
```
Comment (emkookmer) [inline HomeAssistant.js:159]: missing translation : "Preset mode"

File: HomeAssistant.js [hunk 10]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
+        this.publish(this.config.mode_state_topic, data, "Update mode")
     }
 
     // 预设模式
     publish_preset_mode(data) {
-        this.publish(this.config.preset_mode_state_topic, data, "更新预设模式")
+        this.publish(this.config.preset_mode_state_topic, data, "Update preset mode")
     }
 
     // 摆动模式
```
Comment (emkookmer) [inline HomeAssistant.js:169]: missing translation : "Swing Mode"

File: HomeAssistant.js [hunk 11]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_swing_mode(data) {
-        this.publish(this.config.swing_mode_state_topic, data, "更新摆动模式")
+        this.publish(this.config.swing_mode_state_topic, data, "Update swing mode")
     }
 
     // 风速模式
```
Comment (emkookmer) [inline HomeAssistant.js:174]: missing translation : "Wind speed"

**Comment by benfiglin** [comment]
No need to spam with comments on each missing translation. This fork is incomplete, and should not be merged yet.

File: HomeAssistant.js [hunk 12]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_fan_mode(data) {
-        this.publish(this.config.fan_mode_state_topic, data, "更新风速模式")
+        this.publish(this.config.fan_mode_state_topic, data, "Update fan mode")
     }
 
     // 亮度
```
Comment (emkookmer) [inline HomeAssistant.js:179]: missing translation : "Brightness"

File: HomeAssistant.js [hunk 13]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
     publish_brightness(data) {
-        this.publish(this.config.brightness_state_topic, data, "更新亮度")
+        this.publish(this.config.brightness_state_topic, data, "Update brightness")
     }
 
     // 电量
```
Comment (emkookmer) [inline HomeAssistant.js:184]: missing translation : "Battery Level"

File: HomeAssistant.js [hunk 14]
```diff
@@ -104,121 +95,115 @@ module.exports = class {
...
-        this.publish(this.config.error_topic, data, "更新错误状态")
+        this.publish(this.config.error_topic, data, "Update error status")
     }
 
     publish_fan_speed(data) {
-        this.publish(this.config.fan_speed_topic, data, "更新风速")
+        this.publish(this.config.fan_speed_topic, data, "Update fan speed")
     }
 
     // 订阅
```
Comment (emkookmer) [inline HomeAssistant.js:209]: missing translation : "Subscribe to topic"

File: HomeAssistant.js [hunk 15]
```diff
@@ -237,7 +222,7 @@ module.exports = class {
     }
 
     // 发布
```
Comment (emkookmer) [inline HomeAssistant.js:224]: missing translation : "Publish data"

**Review by benfiglin — CHANGES_REQUESTED** [review]
Incomplete work, not ready for PR

File: binary_sensor/binary_sensor.html [hunk 16]
```diff
@@ -36,12 +36,12 @@
 
 </script>
 <script type="text/html" data-help-name="ha-mqtt-binary_sensor">
-    <p>二进制传感器</p>
-    <h3>输入</h3>
+    <p>Binary sensor</p>
+    <h3>enter</h3>
```
Comment (emkookmer) [inline binary_sensor/binary_sensor.html:40]: _**E**_nter

File: camera/camera.html [hunk 17]
```diff
@@ -35,12 +35,12 @@
     });
 </script>
 <script type="text/html" data-help-name="ha-mqtt-camera">
-    <p>MQTT摄像头</p>
-    <h3>输入</h3>
+    <p>MQTT camera</p>
+    <h3>enter</h3>
```
Comment (emkookmer) [inline camera/camera.html:39]: _**E**_nter

File: device_automation/device_automation.html [hunk 18]
```diff
@@ -4,17 +4,17 @@
...
     <div class="form-row">
-        <label for="node-input-name"> 实体名称</label>
-        <input type="text" id="node-input-name" placeholder="实体名称">
+        <label for="node-input-name"> Entity name</label>
+        <input type="text" id="node-input-name" placeholder="Entity name">
     </div>
     <div class="form-row">
-        <label for="node-input-action"> 操作</label>
-        <input type="text" id="node-input-action" placeholder="点击、双击、三连击、长按">
+        <label for="node-input-action"> operate</label>
```
Comment (emkookmer) [inline device_automation/device_automation.html:11]: _**O**_perate

File: device_automation/device_automation.html [hunk 19]
```diff
@@ -4,17 +4,17 @@
...
+        <input type="text" id="node-input-action" placeholder="Click, double click, three courses, long press">
```
Comment (emkookmer) [inline device_automation/device_automation.html:12]: three courses ?? I think you mean triple click

Comment (benfiglin) [reply device_automation/device_automation.html:12]: This is google translate. The work is INCOMPLETE. why did you create a PR??

File: discovery/discovery.html [hunk 20]
```diff
@@ -3,21 +3,21 @@
...
+    <div class="form-tips"> Transmit <b>online</b> to <b>homeassistant/status</b> subscribe node to automatically discover all related devices under MQTT</div>
 </script>
 <script type="text/javascript">
     RED.nodes.registerType('ha-mqtt-discovery', {
-        category: '家庭助理',
+        category: 'Home Assistant',
         color: '#C0DEED',
         icon: "font-awesome/fa-search",
-        paletteLabel: '自动发现',
+        paletteLabel: 'Automatically discover',
```
Comment (emkookmer) [inline discovery/discovery.html:13]: "Automatic discovery" might be a beter translation

File: discovery/discovery.html [hunk 21]
```diff
@@ -3,21 +3,21 @@
...
         defaults: {
             server: { value: "", type: "mqtt-broker", required: true }
         },
         inputs: 1,
         outputs: 0,
         label: function () {
-            return "HomeAssistant发现设备";
+            return "HomeAssistant discovery equipment";
```
Comment (emkookmer) [inline discovery/discovery.html:20]: "Discovered device"  might be a better translation

File: discovery/discovery.js [hunk 22]
```diff
@@ -7,10 +7,10 @@ module.exports = function (RED) {
             const node = this
             node.on('input', function (msg) {
                 node.server.client.publish('homeassistant/status', 'online')
-                this.status({ fill: "green", shape: "ring", text: `${new Date().toLocaleTimeString()} 发现设备` });
+                this.status({ fill: "green", shape: "ring", text: `${new Date().toLocaleTimeString()} Discovery equipment` });
```
Comment (emkookmer) [inline discovery/discovery.js:10]: "Discovered device" might be a better translation

**Comment by benfiglin** [comment]
You really don't read others' comments... do you?

**Comment by emkookmer** [comment]
Hi @benfiglin ,

I do reed comment, but I was using review mode, which doesn't show comment/replies.

Also I know this branch isn't ready yet, but with this I hoped to create insight for everyone interested.

**Comment by benfiglin** [comment]
@emkookmer instead of spamming my email with 30 comments, you could have just created a fork and fixed those translation issues. 

I needed a quick and dirty solution for myself and used google translate to do the translation (hence the obvious translation mistakes). I also changed other things to fit my needs, so this PR will contain many "breaking" changes.
I planned to create a clean PR with a proper translation at some point, but I don't have the time for this right now.

If you have the time, please fork the original and do the fixes since my fork is a breaking change and should not be merged here.

**Comment by emkookmer** [comment]
Hi @benfiglin

The breaking changes you are referring to, might that be the "device" you added?

if so, I think this is a welcome addition

**Comment by benfiglin** [comment]
It is very welcome, but it will break the node for those who update. 
I also changed the default device identifiers which will break the existing devices in HA. 
Finally, I probably made it unusable for the Chinese users since I removed the use of the `pinyin` module.

The PR should focus on one and only thing, which is the translation. Other PRs can focus on things like adding a device definition, etc.

Also, you realize that we take someone's work which was made for the Chinese speaking community and make it unusable for them? I think there needs to be a separate repository for the English version of this addon, since I'm sure some would still like to use this one in Chinese. 
Even better, follow this for internationalization and make both languages available: https://nodered.org/docs/creating-nodes/i18n

**Comment by emkookmer** [comment]
I realize that it was made for the Chinese speaking community, but the same can be said when publishing a library/palette which is used by all other communities. 

Thus internationalization is definitely the way to go.

BTW, thanks for the tip

**Comment by benfiglin** [comment]
I think it should have been made for the default language of Node Red first (en-US) and then add additional languages to it as needed. But since it is already public and used we need to make sure that any changes introduced will not break it for everyone else. 

I might look into the internationalization next week or so, and maybe create a proper, clean PR.

File: device_automation/device_automation.html (hunk 19 above)
Comment (JBaczuk) [reply device_automation/device_automation.html:12]: wow this is funny to read

**Pull request merged**